globus_cli/safeio/write.py

```diff
@@ -15,3 +17,12 @@ def safeprint(message, write_to_stderr=False, newline=True):
             pass
         else:
             raise
+
+
+def print_command_hint(message):
+    """
+    Wrapper around safeprint that checks terminal state
+    before printing a given command hint message
+    """
+    if term_is_interactive() and err_is_terminal():
```

Comment:
I was discussing this more with Jason. I think I was the one who originally suggested this behavior, but it leads to some mild weirdness if you do
`globus session show > output`
because stderr is still the terminal and the hint prints.

I think that checking that `out_is_terminal()` as well would be appropriate.